mlrun/runtimes/sparkjob.py

```diff
@@ -177,19 +177,63 @@ def _run(self, runobj: RunObject, execution: MLClientCtx):
         update_in(job, "spec.driver.volumeMounts", self.spec.volume_mounts)
         update_in(job, "spec.executor.volumeMounts", self.spec.volume_mounts)
         update_in(job, "spec.deps", self.spec.deps)
+        if "limits" in self.spec.resources:
+            if "cpu" in self.spec.resources["limits"]:
+                update_in(
+                    job, "spec.executor.coreLimit", self.spec.resources["limits"]["cpu"]
+                )
         if "requests" in self.spec.resources:
             if "cpu" in self.spec.resources["requests"]:
                 update_in(
                     job, "spec.executor.cores", self.spec.resources["requests"]["cpu"]
                 )
-        if "limits" in self.spec.resources:
-            if "cpu" in self.spec.resources["limits"]:
+            if "memory" in self.spec.resources["requests"]:
                 update_in(
-                    job, "spec.executor.coreLimit", self.spec.resources["limits"]["cpu"]
+                    job,
+                    "spec.executor.memory",
+                    self.spec.resources["requests"]["memory"],
+                )
+            gpu_type = [
+                x for x in self.spec.resources["requests"] if x not in ["cpu", "memory"]
+            ]
+            if gpu_type:
+                update_in(job, "spec.executor.gpu.name", gpu_type[0])
```

Comment:
If for some reason more than one type of gpu was added, you're simply ignoring it
please add an `if len(gpu_type) > 1` that will log some warning so at least we will be able to debug this scenario

Reply:
I'm ignoring it as SparkApplication support only one (https://github.com/GoogleCloudPlatform/spark-on-k8s-operator/blob/6f35a11526257c3ed6d48b68f48e06d9f1e48999/pkg/apis/sparkoperator.k8s.io/v1beta2/types.go#L407)
+ the with_driver/executor_limits do not allow adding more than one anyway.

The question here is how to fail as this code runs on the server side.
I can throw an exception (not sure if it's the common way to go here).
If I don't want to fail, but only warn, my only way to go here is to use runobj.status.status_text which is already used for the UI URL

Reply:
Okay, I see why it basically can't happen, but still please check that for sanity, if length bigger than 1, raise an exception (yes you can do it and it is okay).
Please also use `self.spec.executor_resources` for the executor and don't use `self.spec.resources` for it

Reply:
done